lib/common/ipc_attrd.c

```diff
@@ -497,3 +497,96 @@ pcmk__attrd_api_update(pcmk_ipc_api_t *api, const char *node, const char *name,
 
     return rc;
 }
+
+int
+pcmk__attrd_api_update_list(pcmk_ipc_api_t *api, GList *attrs, const char *dampen,
+                            const char *set, const char *user_name,
+                            uint32_t options)
+{
+    int rc = pcmk_rc_ok;
+    xmlNode *request = NULL;
+
+    if (attrs == NULL) {
+        return EINVAL;
+    }
+
+    /* There are two different ways of handling a list of attributes:
+     *
+     * (1) For messages originating from some command line tool, we have to send
+     *     them one at a time.  In this loop, we just call pcmk__attrd_api_update
+     *     for each, letting it deal with creating the API object if it doesn't
+     *     already exist.
+     *
+     *     The reason we can't use a single message in this case is that we can't
+     *     trust that the server supports it.  Remote nodes could be involved
+     *     here, and there's no guarantee that the version of attrd running on a
+     *     remote node will be recent enough.  We also can't just try sending a
+     *     single message and then falling back on multiple.  There's no handshake
+     *     with the attrd server to determine its version.  And then we would
+     *     need to do that fallback in the dispatch function for this to work for
+     *     all connection types (mainloop in particular), and at that point we
+     *     won't know what the original message was in order to break it apart
+     *     and resend as individual messages.
+     *
+     * (2) For messages between daemons, we can be assured that everything will
+     *     have the right version and that a single big message is supported.  So,
```

Comment:
more specifically, we can be sure that the local attrd will support the new message, and it can send it to the other attrds as one request or split up according to the minimum supported version

not important, but fyi: pacemaker-remoted itself (which is a flavor of pacemaker-execd) would be in the same boat as a remote client, but it doesn't talk to attrd so it doesn't matter

Reply:
Fixed.